libr/util/buf.c

```diff
@@ -48,7 +49,7 @@ static RBufferSparse *sparse_append(RList *l, ut64 addr, const ut8 *data, int le
 	s->size = len;
 	s->odata = NULL;
 	s->data = calloc (1, len);
-	if (!data) {
+	if (!s->data) {
 		free (s);
 		return NULL;
 	} else {
```

Comment:
no need for the `else` here since there is the return NULL in the if branch and it makes the code more readable.